app/views/display_ads/show.html.erb

```diff
@@ -0,0 +1,9 @@
+<% if @display_ad %>
+  <% if user_signed_in? %>
+    <%= render partial: "shared/display_ad", locals: { display_ad: @display_ad, data_context_type: DisplayAdEvent::CONTEXT_TYPE_ARTICLE } %>
+  <% else %>
+    <% cache([params[:username], params[:slug], params[:placement_area]], expires_in: 15.minutes) do %>
```

Comment:
More of a feature question: so we don't want this to be based on article?

Reply:
I don't think we need it to be based on the article attributes, I think the slug of an article can be sufficient to determine a cache entry.